Mathlib/Topology/MetricSpace/HausdorffDistance.lean

```diff
@@ -544,6 +544,10 @@ theorem _root_.IsClosed.not_mem_iff_infDist_pos (h : IsClosed s) (hs : s.Nonempt
     x ∉ s ↔ 0 < infDist x s := by
   simp [h.mem_iff_infDist_zero hs, infDist_nonneg.gt_iff_ne]
 
+theorem infDist_pos_iff_not_mem_closure (hs : s.Nonempty) :
+    0 < infDist x s ↔ x ∉ closure s := by
+  rw [isClosed_closure.not_mem_iff_infDist_pos hs.closure, infDist_closure]
```

Comment:
Alternative proof you could put right next to `mem_closure_iff_infDist_zero`
```suggestion
    0 < infDist x s ↔ x ∉ closure s :=
  infDist_nonneg.gt_iff_ne.trans (mem_closure_iff_infDist_zero hs).not.symm
```
Note that the three similar lemmas have the mem on the LHS

Reply:
Moved and flipped.